### ISSUE TITLE
Incorrect nullable warning under `nameof` expression which cannot be easily suppressed

### ISSUE THREAD

**Issue by TessenR** [dotnet/roslyn#31436]
**Version Used**: 

Branch C# 8.0: Nullable reference type (29 Oct 2018)
https://github.com/dotnet/csharplang/blob/master/proposals/nullable-reference-types.md
Latest commit 2f8fef by AlekseyTs:
Compare Nullable modifiers while comparing type symbols by default. (#30770)

**Steps to Reproduce**:

Compile the following code
```cs
#nullable enable
public class C {
    public static void Main() => M(null);
    static void M(C? c)
    {
      var s = nameof(c.ToString);
    }
}
```

https://sharplab.io/#v2:EYLgZgpghgLgrgJwgZwLQDk4BstWFiAJQkiQDsBjCAFQE8AHFAGhhATjIB8BiM7XfBAAEEMngIBYAFAABAMxCZAJiEBhIQG9pQnYoUyAjADZFAFiEBZKAEsyACgCUQgLwA+S3b44HAbm27DExlzCztVAH4hCgd/HS0pXV0ANygEIWQXITEAWwgAezA7CgA6ajyAZRgEWwBzX1ihAF9pRqA==

**Expected Behavior**:

No `warning CS8602: Possible dereference of a null reference.` since `nameof` doesn't actually execute its argument and there's no dereference at all.

**Actual Behavior**:

`warning CS8602: Possible dereference of a null reference` is reported for `nameof(c.ToString())`

**Notes**

There's also no easy way to get rid of this false warning since both `nameof(c?.ToString)` and `nameof(c!.ToString)` result in compiler errors

**Comment by chsienki** [comment]
No repro on master as of 9cb17ff35b
